### PR TITLE
Add post-processing

### PR DESCRIPTION
**Pull request by SunshineJohn**
Added post-processing to the renderer. Partially refactored to accept const char* instead of const std::string& as a parameter to functions.

### CODE REVIEW

File: Code/Renderer/ShaderProgram.cpp [hunk 1]
```diff
@@ -66,64 +68,84 @@ bool ShaderProgram::UseProgram() const
     return false;
 }
 
-void ShaderProgram::SetBool(const std::string& name, bool value) const
+void ShaderProgram::SetBool(const char* name, bool value) const
 {
-    glUniform1i(glGetUniformLocation(program, name.c_str()), (int)value);
+    glUniform1i(glGetUniformLocation(program, name), (int)value);
```
Comment (KaterynaMelnyk) [inline Code/Renderer/ShaderProgram.cpp:73]: why don't you use static_cast in `(int)value` ?

File: Code/Renderer/ShaderProgram.cpp [hunk 2]
```diff
@@ -12,14 +14,14 @@ ShaderProgram::~ShaderProgram()
...
-void ShaderProgram::SetVertexShader(const std::string& vertexPath)
+void ShaderProgram::SetVertexShader(const char* vertexPath)
 {
     vertexShader.ChangeFileName(vertexPath);
     vertexShader.ChangeType(GL_VERTEX_SHADER);
     vertexShader.Compile();
 }
 
-void ShaderProgram::SetFragmentShader(const std::string& fragmentPath)
+void ShaderProgram::SetFragmentShader(const char* fragmentPath)
```
Comment (KaterynaMelnyk) [inline Code/Renderer/ShaderProgram.cpp:24]: Should there be `const`?

File: Code/Renderer/ShaderProgram.cpp [hunk 3]
```diff
@@ -12,14 +14,14 @@ ShaderProgram::~ShaderProgram()
     glDeleteProgram(program);
 }
 
-void ShaderProgram::SetVertexShader(const std::string& vertexPath)
+void ShaderProgram::SetVertexShader(const char* vertexPath)
```
Comment (KaterynaMelnyk) [inline Code/Renderer/ShaderProgram.cpp:17]: Should there be `const` ?

File: Code/Renderer/ShaderProgram.h [hunk 4]
```diff
@@ -13,29 +13,32 @@ class ShaderProgram
...
     ShaderProgram() {}
-    ShaderProgram(const std::string& vertexPath,
-                  const std::string& fragmentPath);
+    ShaderProgram(const char* vertexPath,
+                  const char* fragmentPath);
     ~ShaderProgram();
 
-    void SetVertexShader(const std::string& vertexPath);
-    void SetFragmentShader(const std::string& fragmentPath);
+    void SetVertexShader(const char* vertexPath);
```
Comment (KaterynaMelnyk) [inline Code/Renderer/ShaderProgram.h:20]: should there be `const` ?

File: Code/Renderer/ShaderProgram.cpp (hunk 2 above)
Comment (SunshineJohn) [reply Code/Renderer/ShaderProgram.cpp:24]: Yes, I am not supposed to be able to change the variable value. (Same for the comments below :-) )

**Pull request merged**